src/hazmat.rs

```diff
@@ -64,18 +68,26 @@ impl ExpandedSecretKey {
         bytes
     }
 
-    /// Construct an `ExpandedSecretKey` from an array of 64 bytes.
+    /// Construct an `ExpandedSecretKey` from an array of 64 bytes. In the spec, the bytes are the
+    /// output of a SHA-512 hash. This clamps the first 32 bytes and uses it as a scalar, and uses
+    /// the second 32 bytes as a domain separator for hashing.
+    #[allow(clippy::unwrap_used)]
     pub fn from_bytes(bytes: &[u8; 64]) -> Self {
         // TODO: Use bytes.split_array_ref once it’s in MSRV.
-        let mut lower: [u8; 32] = [0u8; 32];
-        let mut upper: [u8; 32] = [0u8; 32];
+        let (lower, upper) = bytes.split_at(32);
+
+        // Convert the slices to [u8; 32]
+        let scalar_bytes = lower.try_into().unwrap();
+        let hash_prefix = upper.try_into().unwrap();
```

Comment:
I think it'd be nice to eventually turn on the `clippy::unwrap_used` lint.

FWIW, a [quick check in godbolt](https://godbolt.org/z/sTYd48nGK) shows bounds checks are elided here.